src/main/vcp_hal/usbd_cdc_interface.c

```diff
@@ -206,7 +206,7 @@ static int8_t CDC_Itf_Control (uint8_t cmd, uint8_t* pbuf, uint16_t length)
     break;
 
   case CDC_GET_LINE_CODING:
```

Comment:
type of `LineCoding` is `LINE_CODING`, this change shouldn't be necessary and it's better to use sizeof on values rather than typos to avoid multiple sources of authority

Reply:
LineCoding is of type USBD_CDC_LineCodingTypeDef which isn't packed whereas LINE_CODING is packed. The bug was that CDC_GET_LINE_CODING was failing because the buffer length was being compared to the size of the unpacked, not packed structure. I find checking against sizeof(<type>) to be easier to read, but admittedly this will break if the type of the structure in question is changed. I'll change it.

Reply:
Yes, you're right here, but not in other places.